### ISSUE TITLE
Homey app keeps crashing due to CPU warning limit reached

### ISSUE THREAD

**Issue by ApacheCox** [RobinBol/io.everythingsmart#11]
ive had this app installed for a long time but at to reinstall it recently and ever since its been crashing every few mins 

homey dianostic code: 72088c39-afb5-4fe2-b34b-3e0d71e2f6eb

![Skærmbillede 2024-09-21 155527](https://github.com/user-attachments/assets/b32e3730-91f7-4bbd-97e9-dd261439e555)

**Comment by edmond88888** [comment]
Unfortunately I have the same issue. please fix it 🙏 

Diagnostics code : 80db7936-5c87-4009-9c4a-7e90b05742f6

**Comment by edmond88888** [comment]
It seems that when only 2 Everything Presence Lite  sensors are running the app is working properly. 

Only when there are 3 or more Everything Presence Lite sensors are connected the app is crashing. 

@ApacheCox, can you check if you are experiencing the same? 

@RobinBol, can you check what is causing this crash?

**Comment by ApacheCox** [comment]
I got 2 presence one and 1 lite connected currently, but I will check if removing any one of them fixes the issue tonight

**Comment by ApacheCox** [comment]
So I've been testing for about an hour now and it seem you're right having more than two sensors added makes the app crash no matter the configuration ( lites or ones ), but I've also noticed that two of my sensors are showing motion constantly which might be an explanation

**Comment by edmond88888** [comment]
I don’t have the issue with constantly motion

it would be nice if @RobinBol could give us a feedback how to solve the issue with more than 3 EP1/Lites.

**Comment by ptahbe** [comment]
I also have the same issue. The app crashes every few minutes. But at the moment I only have 2 EP1's.

Diagnostics code : f9c682ee-5938-4f74-9491-8c300b76850e

**Comment by edmond88888** [comment]
That’s odd…. 

And if you disconnected one of them?

**Comment by ptahbe** [comment]
I removed 1 EP1 from Homey and at 1st sight the app doesn't crash anymore. But it has only been 10 minutes now.

**Comment by edmond88888** [comment]
When I connect 3 or more EP the app will crash between 45 sec and 2 minutes. Also will be down for 120 seconds before reconnecting again.

**Comment by ApacheCox** [comment]
I have the same issue as @ptahbe out of the four sensors i own only one can be connected to Homey without the app crashing but every 2 mins seems about right

**Comment by edmond88888** [comment]
Hi All, do you still have this issue?

**Comment by ApacheCox** [comment]
Yes still only able to have one sensor connected, still having the constant motion on two of them and a fourth one Homey doesn't even detect even though it very much is on the network. How about you guys?

**Comment by edmond88888** [comment]
The past few days I had only two EP lites online. No issues at all.

Today I have added another EP lite. Up until now (2hours) the app did not crash yet. Hopefully this stays working. Tomorrow I will be adding the fourth EP lite. I will update you….

**Comment by ApacheCox** [comment]
Sounds promising

**Comment by ptahbe** [comment]
At the moment I have 1 EP1 connected, no Lite. There are 2 EP1 that are online, but not connected. I wanted to re-add an EP1, but Homey says there are no devices.

**Comment by edmond88888** [comment]
I was celebrating too fast. After a few hours the app was crashing again ☹️

I have disconnected the third EP lite and after that it was running again.

The motion is doing normal in my situation. 

I can connect all of them, but need to activate wifi through home assistant first (only the first time)

**Comment by ApacheCox** [comment]
It's been over a month now and this issue still persists, even went to a friend's house who uses HA to see if my sensors was the problem but they worked perfectly

**Comment by RobinBol** [comment]
Thanks for the information. Unfortunately I can't really fix this as I currently don't have enough EP sensors to trigger this issue and the CPU limitation is set by Homey.

I did take a look at the code and made some optimizations which may or may not help to address this issue. Would you try to install the [v1.1.4 test version](https://homey.app/nl-nl/app/io.everythingsmart/Everything-Smart/test/) and let me know if this update made any changes to how much devices you can connect simultaneously and for how long?

**Comment by cgiacca** [comment]
Hi, I've installed the test version and added a third sensor. So far it seems OK. I'll keep an eye on it through the day and let you know if it starts to crash again. I have 1 x Presence One and 2 x Presence Lite sensors.

If it's OK, I'll add my fourth sensor as well (another Presence Lite).

**Comment by cgiacca** [comment]
Woke up this morning to find the app had crashed with 3 sensors. It lasted around 48 hrs. I've removed the third sensor and will just leave two connected.

While I appreciate your efforts and understand it may be a limitation of the Homey itself, I'd recommend putting a disclaimer on the app description stating that it becomes unstable if more than two sensors are added. It would have saved me buying 3 sensors I cannot use.

**Comment by ApacheCox** [comment]
@cgiacca I've used this app for at least a year using 4 sensors without issues, it was only back in September the issues started so I don't think it's a hardware limitation of the homey pro

**Comment by cgiacca** [comment]
I was just going by the earlier statement that the CPU limitation is set by Homey. So maybe not hardware, but a software limitation.
